### PR TITLE
[Backend] Add or delete challenge phases and splits

### PR DESCRIPTION
**Pull request by Suryansh5545**
removes error messages from the validation process and adds ability to add or remove challenge phase , phase split, datasplit

https://github.com/Cloud-CV/EvalAI/assets/34577232/c49e72c4-aea7-4f9f-bebf-2b4d9e08ec71

### CODE REVIEW

File: apps/challenges/views.py [hunk 1]
```diff
@@ -3801,7 +3813,6 @@ def create_or_update_github_challenge(request, challenge_host_team_pk):
                     ).first()
                     if (
                         challenge_test_annotation_file
-                        and not challenge_phase.annotations_uploaded_using_cli
```
Comment (gchhablani) [inline apps/challenges/views.py:3804]: What was this thing again? Can you please elaborate why this change was needed?

**Review by gchhablani — COMMENTED** [review]
If you allow adding/deleting challenge phases, you also need to look into challenge phase splits. Challenge phase splits are dependent on challenge phases.

Have provided some other comments. Add a check for submissions in all cases. If there are related submissions, we should not allow deleting anything.

File: apps/challenges/views.py (hunk 1 above)
Comment (Suryansh5545) [reply apps/challenges/views.py:3804]: Adding the challenge phase split after creation of challenge gives a error NoneType for annotations_uploaded_using_cli. we do not check for this in the creation code, there is another check below this if statement which does the opposition and check if it exist or not for annotations_uploaded_using_cli

File: apps/challenges/views.py [hunk 2]
```diff
@@ -3782,6 +3782,18 @@ def create_or_update_github_challenge(request, challenge_host_team_pk):
...
                 challenge_phases_data = yaml_file_data["challenge_phases"]
+
+                # Delete the challenge phase if it is not present in the yaml file
+                existing_challenge_phases = ChallengePhase.objects.filter(challenge=challenge)
+                existing_challenge_phase_ids = [str(challenge_phase.config_id) for challenge_phase in existing_challenge_phases]
+                challenge_phases_data_ids = [str(challenge_phase_data["id"]) for challenge_phase_data in challenge_phases_data]
+                challenge_phase_ids_to_delete = list(set(existing_challenge_phase_ids) - set(challenge_phases_data_ids))
+                for challenge_phase_id_to_delete in challenge_phase_ids_to_delete:
+                    challenge_phase = ChallengePhase.objects.filter(
+                        challenge__pk=challenge.pk, config_id=challenge_phase_id_to_delete
```
[the remark this answers is not in the document]
Comment (Suryansh5545) [reply apps/challenges/views.py:3833]: done

**Comment by Suryansh5545** [comment]
Added ability to add and remove challange phase split, merger depends on #4061.

https://github.com/Cloud-CV/EvalAI/assets/34577232/f5cfd48f-baa5-4ce4-9b50-7e77096f8dd6

**Comment by gchhablani** [comment]
@Suryansh5545 Remove the error messages that are not being used anymore.
